### DISCUSSION TITLE
How to use integrate function

### DISCUSSION THREAD

**Discussion by n-flock** [facontidavide/PlotJuggler#554] [Q&A]
New to using Plotjuggler.  Trying to see how much energy a motor consumes over time.  Have instantaneous power plotted.  Wan to sum power over time.  Tried using "apply filter to data"/integral and got a curve that is shaped as expected but y-axis values don't make sense --> have 6-7 extra orders of magnitude from expectation.  Is there a more detailed user guide available?  Have gone through the 3 tutorials on github.  Haven't been able to find anything more detailed on the web.

**Comment by facontidavide** [comment]
Hi,

if you think that the integral solution is really wrong, please submit an issue and I will fix it (with a proper datset and instruction to repdoruce your problem).

If you need a specific formula, just use the Custom Plots and the Lua editor and create your own formula.

**Reply by n-flock** [reply]
Here is the plot:

https://user-images.githubusercontent.com/93461195/139789934-2de2fe63-18dc-4b14-adba-8c4a74f45798.png

Third chart contains the integral filter.  Second chart is product of the data shown in first chart (current * voltage).  Resulting instantaneous power in the 700-1000W range is expected.  Integral should give me energy consumption.  Battery is only 672Wh rated (and I don't really trust it gets the claimed range), so the integrated curve going up past 1.5e12 doesn't match expectations.

Maybe the 1e5 multiplier on my time series is the issue (sampling at 10Hz/displaying as 1e5 1e5 μs).  Dividing by 1e5 to get from μs to 0.1s, another 10 to get to seconds, and 3600 to get to hours gives 416.67, which is at least in the ballpark, though that means the battery is only providing 2/3 of rated energy and has only done maybe a dozen cycles.  I guess next step will be to run a load test on the battery.
